src/lib.rs

```diff
@@ -561,6 +524,53 @@ impl<'text> BidiInfo<'text> {
     }
 }
 
+
+/// Contains a reference of `Bidi_info` and one of its `paragraphs`.
+#[derive(Debug)]
+pub struct Paragraph<'a, 'text> {
+  info: &'a BidiInfo<'text>,
+  para: &'a ParagraphInfo,
+} 
+
+impl <'a, 'text>Paragraph <'a, 'text>{
```

Comment:
We should have ways to construct this from an iterator. I can add them if you'd like since iterators are a bit tricky.

Reply:
If it is okay, I can add this in a following PR.

Reply:
okay